packages/embedder/ai_models/AbstractModel.ts

```diff
@@ -42,7 +48,68 @@ export abstract class AbstractEmbeddingsModel extends AbstractModel {
     this.tableName = `Embeddings_${modelParams.tableSuffix}`
   }
   protected abstract constructModelParams(config: EmbeddingModelConfig): EmbeddingModelParams
-  abstract getEmbedding(content: string): Promise<number[]>
+  abstract getEmbedding(content: string): Promise<number[] | Error>
+
+  abstract getTokens(content: string): Promise<number[] | Error>
+
+  async createTable() {
+    const pg = getKysely()
+    const hasTable =
+      (
+        await sql<number[]>`SELECT 1 FROM ${sql.id('pg_catalog', 'pg_tables')} WHERE ${sql.id(
+          'tablename'
+        )} = ${this.tableName}`.execute(pg)
+      ).rows.length > 0
+    if (hasTable) return undefined
+    const vectorDimensions = this.embeddingDimensions
+    console.log(`ModelManager: creating ${this.tableName} with ${vectorDimensions} dimensions`)
+    await sql`
+      DO $$
+        BEGIN
+        CREATE TABLE IF NOT EXISTS ${sql.id(this.tableName)} (
+          "id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+          "embedText" TEXT,
+          "embedding" vector(${sql.raw(vectorDimensions.toString())}),
+          "embeddingsMetadataId" INTEGER UNIQUE NOT NULL,
```

Comment:
@jordanh safe to add the unique constraint here?

Reply:
Yes, I think that is ok. It would imply that if we ever have multiple versions of the same embedding models, they'll need to live on their own table which I think is what we'd want